src/main/java/org/flickit/flickitassessmentcore/adapter/out/persistence/assessment/AssessmentJpaRepository.java

```diff
@@ -40,9 +40,12 @@ void update(@Param(value = "id") UUID id,
 
     @Modifying
     @Query("UPDATE AssessmentJpaEntity a SET " +
-        "a.deletionTime = :deletionTime " +
+        "a.deletionTime = :deletionTime, " +
+        "a.deleted = true " +
         "WHERE a.id = :id")
-    void setDeletionTimeById(@Param(value = "id") UUID id, @Param(value = "deletionTime") Long deletionTime);
+    void setDeletedAndDeletionTimeById(@Param(value = "id") UUID id, @Param(value = "deletionTime") Long deletionTime);
+
+    boolean existsByIdAndDeleted(@Param(value = "id") UUID id, @Param(value = "deleted") boolean deleted);
```

Comment:
can be changed to `boolean existsByIdAndDeletedFalse(@Param(value = "id") UUID id);`